.github/workflows/ci_tests.yml

```diff
@@ -74,31 +67,41 @@ jobs:
           ninjaVersion: "^1.11.1"
       - name: Print installed softwares
         run: |
-          clang++ --version
-          g++ --version
+          ${{ matrix.compiler.cpp }} --version || echo ignored
+          ${{ matrix.compiler.c }} --version || echo ignored
           cmake --version
           ninja --version
+
+      - name: Configure CMake on macos with clang
+        if: startsWith(matrix.compiler.c, 'clang') && startsWith(matrix.platform, 'macos')
+        run: |
+          CC=$(brew --prefix llvm@18)/bin/clang CXX=$(brew --prefix llvm@18)/bin/clang++ cmake -B build -S . -DCMAKE_CXX_STANDARD=${{ matrix.cpp_version }} ${{ matrix.cmake_args.args }}
+        env:
+          CMAKE_GENERATOR: "Ninja Multi-Config"
+
       - name: Configure CMake
+        if: ( startsWith(matrix.compiler.c, 'gcc') || startsWith(matrix.platform, 'ubuntu') ) || ( startsWith(matrix.compiler.c, 'gcc') && startsWith(matrix.platform, 'macos') )
         run: |
           cmake -B build -S . -DCMAKE_CXX_STANDARD=${{ matrix.cpp_version }} ${{ matrix.cmake_args.args }}
         env:
           CC: ${{ matrix.compiler.c }}
           CXX: ${{ matrix.compiler.cpp }}
           CMAKE_GENERATOR: "Ninja Multi-Config"
+
       - name: Build Release
         run: |
           cmake --build build --config Release --verbose
           cmake --build build --config Release --target all_verify_interface_header_sets
-          cmake --install build --config Release --prefix /opt/beman.exemplar
-          find /opt/beman.exemplar -type f
+          cmake --install build --config Release --prefix /tmp/beman.exemplar
+          find /tmp/beman.exemplar -type f
```

Comment:
Any reason not to keep /opt? It's usually a path used for Linux libraries and I think that path should also exist for MAC OS.

Reply:
Yes, but `/opt` is owned by root.
I would use `${PWD}/stagedir` like boost does.

Reply:
I think we supposed regardless of who is owner of /opt, all users can create a subdirectory or whatever.

My point is that /opt is used in all repos in Beman right now in documentation. If this recommendation is not OK, I'm OK to change it everywhere (other PRs for docs), but if it's works, I won't do the change.

I will also test and come back.

Reply:
### from CI build on MAC OS:
```bash
ld: warning: ignoring duplicate libraries: 'lib/Release/libgtest.a'
[1/1] Building CXX object src/beman/exemplar/CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/Release/beman.exemplar_verify_interface_header_sets/beman/exemplar/identity.hpp.cxx.o
CMake Error at build/src/beman/exemplar/cmake_install.cmake:44 (file):
  file cannot create directory: /opt/beman.exemplar/include/beman/exemplar.
  Maybe need administrative privileges.
Call Stack (most recent call first):
  build/cmake_install.cmake:42 (include)


Error: Process completed with exit code 1.
```

Reply:
```bash
dariusn@pc:~/git/Beman/iterator_interface$ ls -l / | grep opt
drwxr-xr-x   7 dariusn root       4096 Nov 12 07:38 opt
```
I guess it depends if and how you have configured `/opt`.  So we may add:
```
# Create the /opt directory if it doesn’t already exist
sudo mkdir -p /opt

# Change ownership of /opt to the current user
sudo chown -R $USER /opt
```
(check other existing commands, sudo may not be required!)


I would keep the current PR just for adding MAC OS support. We can discuss on discourse (https://discourse.bemanproject.org/latest) in a new post if `/opt` is or not the best path to recommend to install in docs, etc CI.

We should aim to have the same default path in all places/repos! (docs, CI files, etc). It just makes the usability simpler for new users! And it still allows folks to use other custom install path on own system!

Reply:
It does not successfully run on **Darwin** if we install to `/opt`?

As a user of project, I want to have control if and where the binaries are installed!

I am not willing to change the.

Reply:
/opt is the correct location in a unix FHS style system because every other prefix is owned by someone else. / and /usr (now fused anyway) are owned by the OS distro, /usr/local is the machine admin, ~ and ~/.local are user controlled.
https://www.pathname.com/fhs/pub/fhs-2.3.html#OPTADDONAPPLICATIONSOFTWAREPACKAGES

"A package to be installed in /opt must locate its static files in a separate /opt/<package> or /opt/<provider> directory tree, where <package> is a name that describes the software package and <provider> is the provider's LANANA registered name."

So a todo to register `beman` with LANANA, although it's actually unlikely real world installs go there. 



Testing install in CI is also not locking in an install location for any user. PREFIX is controlled by the installer. 

That all said, outside of a scratch VM like a GitHub action runs in, using a directory inside out root, but outside the source tree, like we do for build, is the friendly way to do it. 

Staging isn't the right name, because it's not going anywhere after that. Trying to do a package manager's job is a mistake

Reply:
> As a user of project, I want to have control if and where the binaries are installed!
> 
> I am not willing to change more.
@ClausKlein , I agree with you, as a user you can install them anywhere you want. As a developer inside the Beman project, you must use some requirements/recommendations/best practices used inside the Beman org. That's not questionable, but you are more than welcome to propose to change / upgrade/ improves these sets of standards or practices.


TLDR:
* This CI config is exclusively for internal development. It must follow our internal practices or rules.
* Similarly, examples for root README.md, are just examples and our recommendations.  Our choice, but not an obligation for users.
* Any user can install our libraries in a custom path, that's for sure an available option!

Steve also explained this in previous comment - "Testing install in CI is also not locking in an install location for any user. PREFIX is controlled by the installer."

I hope that is clear now. Please revert the changes related to the install path. Thanks!